From 801c957651fba94b5d7dde9cc490378a4dc3a987 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Thu, 1 Jun 2017 11:40:02 +0200
Subject: [PATCH] [Backport] GN: Add Arm64 Linux support

Add Arm64 linux support be able to build GN binary for
native arm64 builds.

R=dpranke@chromium.org, brettw@chromium.org

Review-Url: https://codereview.chromium.org/2550673002

Task-number: QTBUG-61128
Change-Id: I3aa6b6ca7df374428a7f4a8b4db9f488a8926f6c
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
---
 chromium/tools/gn/args.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/3rdparty/chromium/tools/gn/args.cc b/src/3rdparty/chromium/tools/gn/args.cc
index 7285d7a..54703e1 100644
--- a/src/3rdparty/chromium/tools/gn/args.cc
+++ b/src/3rdparty/chromium/tools/gn/args.cc
@@ -266,6 +266,7 @@ void Args::SetSystemVarsLocked(Scope* dest) const {
   static const char kX86[] = "x86";
   static const char kX64[] = "x64";
   static const char kArm[] = "arm";
+  static const char kArm64[] = "arm64";
   static const char kMips[] = "mipsel";
   static const char kS390X[] = "s390x";
   static const char kPPC64[] = "ppc64";
@@ -280,6 +281,8 @@ void Args::SetSystemVarsLocked(Scope* dest) const {
     arch = kX64;
   else if (os_arch.substr(0, 3) == "arm")
     arch = kArm;
+  else if (os_arch == "aarch64")
+    arch = kArm64;
   else if (os_arch == "mips")
     arch = kMips;
   else if (os_arch == "s390x")
-- 
2.7.4

